### ISSUE TITLE
yii\filters\RateLimiter 

### ISSUE THREAD

**Issue by sanwv** [yiisoft/yii2#7683]
RateLimiter default need a user implement RateLimitInterface,this filter is default configed by yii\rest\Controller
## if restfull api neednt a user this will throw a exception like 

<response>
<name>Invalid Configuration</name>
<message>User::identityClass must be set.</message>
<code>0</code>
<type>yii\base\InvalidConfigException</type>
<file>
/var/www/test/vendor/yiisoft/yii2/web/User.php
</file>

**Comment by RSalo** [comment]
I have the same problem, Maybe by default disable RateLimit for rest controller?

**Comment by samdark** [comment]
Probably yes.

**Comment by samdark** [comment]
@RSalo what's the problem with implementing RateLimitInterface?

**Comment by RSalo** [comment]
@samdark i have config in module
```
return [
    'components' => [
        'user' => [
            'class' => \app\modules\v1\models\Token::className(),
            'identityClass' => \app\modules\v1\models\Token::className(),
            'enableAutoLogin' => false
        ],
    ]
];
```
And get error in authenticator and rateLimiter:
```
{
  "errors": [
    {
      "name": "Invalid Configuration",
      "message": "User::identityClass must be set.",
      "code": 0,
      "type": "yii\\base\\InvalidConfigException",
      "file": "C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\web\\User.php",
      "line": 163,
      "stack-trace": [
        "#0 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\base\\Object.php(107): yii\\web\\User->init()",
        "#1 [internal function]: yii\\base\\Object->__construct(Array)",
        "#2 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\di\\Container.php(375): ReflectionClass->newInstanceArgs(Array)",
        "#3 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\di\\Container.php(156): yii\\di\\Container->build('yii\\\\web\\\\User', Array, Array)",
        "#4 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\BaseYii.php(344): yii\\di\\Container->get('yii\\\\web\\\\User', Array, Array)",
        "#5 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\di\\ServiceLocator.php(135): yii\\BaseYii::createObject(Array)",
        "#6 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\web\\Application.php(186): yii\\di\\ServiceLocator->get('user')",
        "#7 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\filters\\RateLimiter.php(70): yii\\web\\Application->getUser()",
        "#8 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\base\\ActionFilter.php(75): yii\\filters\\RateLimiter->beforeAction(Object(app\\modules\\v1\\controllers\\actions\\token\\IndexAction))",
        "#9 [internal function]: yii\\base\\ActionFilter->beforeFilter(Object(yii\\base\\ActionEvent))",
        "#10 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\base\\Component.php(545): call_user_func(Array, Object(yii\\base\\ActionEvent))",
        "#11 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\base\\Controller.php(272): yii\\base\\Component->trigger('beforeAction', Object(yii\\base\\ActionEvent))",
        "#12 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\web\\Controller.php(164): yii\\base\\Controller->beforeAction(Object(app\\modules\\v1\\controllers\\actions\\token\\IndexAction))",
        "#13 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\base\\Controller.php(154): yii\\web\\Controller->beforeAction(Object(app\\modules\\v1\\controllers\\actions\\token\\IndexAction))",
        "#14 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\base\\Module.php(523): yii\\base\\Controller->runAction('index', Array)",
        "#15 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\web\\Application.php(102): yii\\base\\Module->runAction('v1/token/index', Array)",
        "#16 C:\\OpenServer\\domains\\hochu-backend.loc\\vendor\\yiisoft\\yii2\\base\\Application.php(380): yii\\web\\Application->handleRequest(Object(yii\\web\\Request))",
        "#17 C:\\OpenServer\\domains\\hochu-backend.loc\\index.php(11): yii\\base\\Application->run()",
        "#18 {main}"
      ]
    }
  ]
}
```
But if i do:
```
            'rateLimiter' => [
                'class' => RateLimiter::className(),
                'user' => \Yii::createObject(Token::className()) // <- this
            ],
```
It works well.

Why rate limit and authenticator doesn't take "user" from the config by default?

**Comment by samdark** [comment]
So do I understand correctly that you have two user components. In the application it's without rate limiter, in module it is with rate limiter?

**Comment by samdark** [comment]
Seems currently you have to explicitly set `user` property of the `RateLimiter` and `AuthMethod`:
https://github.com/yiisoft/yii2/blob/4aa935e69e7546d731bf4bafb736243d4abfc5d7/framework/filters/auth/AuthMethod.php#L29. If it's not set then it's asked directly from the application's `user` w/o checking module: https://github.com/yiisoft/yii2/blob/4aa935e69e7546d731bf4bafb736243d4abfc5d7/framework/filters/auth/AuthMethod.php#L58

**Comment by samdark** [comment]
I think we can fall back to currently active module first in this case.

**Comment by RSalo** [comment]
It doesn't work as minimum for RateLimit
https://github.com/yiisoft/yii2/blob/4aa935e69e7546d731bf4bafb736243d4abfc5d7/framework/filters/RateLimiter.php#L70
for AuthMethod
https://github.com/yiisoft/yii2/blob/4aa935e69e7546d731bf4bafb736243d4abfc5d7/framework/filters/auth/AuthMethod.php#L58
 and its all subclasses
yii\filters\auth\CompositeAuth, yii\filters\auth\HttpBasicAuth, yii\filters\auth\HttpBearerAuth, yii\filters\auth\QueryParamAuth

**Comment by samdark** [comment]
- #5141
- #4793
- #9018
- #8556

**Comment by developeruz** [comment]
It seems that issue solved by #13994
The [test](https://github.com/yiisoft/yii2/blob/1453eeaa4a044986e665d2f8406cdf04bd746ee3/tests/framework/filters/RateLimiterTest.php#L100) proves that RateLimiter could be configurated without User class.
